### PR TITLE
allow to query DB stall status

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

This PR adds a new property `is-write-stalled` to query whether the column family is in write stall or not.

### Why it's necessary?
In TiKV there is a compaction filter used for GC, in which `DB::write` is called. So if we can query whether the DB instance is stalled or not, we can skip to create more compaction filter instances to save some resources.

### CODE REVIEW

File: include/rocksdb/utilities/stackable_db.h [hunk 1]
```diff
@@ -415,6 +415,8 @@ class StackableDB : public DB {
     return db_->DefaultColumnFamily();
   }
 
+  virtual bool StallOccurs() const override { return db_->StallOccurs(); }
```
Comment (tabokie) [inline include/rocksdb/utilities/stackable_db.h:418]: maybe `WriteIsStalled`?

File: db/write_thread.h [hunk 2]
```diff
@@ -376,6 +376,11 @@ class WriteThread {
   // Remove the dummy writer and wake up waiting writers
   void EndWriteStall();
 
+  bool StallOccurs() const {
```
Comment (tabokie) [inline db/write_thread.h:379]: `IsStalled`

**Review by tabokie — COMMENTED** [review]
Mind explain why we need this specifically? And can that be fulfilled with a callback instead?

**Comment by yiwu-arbug** [comment]
Can we implement this as a DB property? That would avoid adding the new DB method.

File: db/internal_stats.cc [hunk 3]
```diff
@@ -874,6 +880,12 @@ bool InternalStats::HandleIsWriteStopped(uint64_t* value, DBImpl* db,
   return true;
 }
 
+bool InternalStats::HandleIsWriteStalled(uint64_t* value, DBImpl* /*db*/,
+                                         Version* /*version*/) {
+  *value = cfd_->IsStalled() ? 1 : 0;
```
Comment (yiwu-arbug) [inline db/internal_stats.cc:885]: To be consistent with `HandleIsWriteStopped`:
```
*value = db->write_controller().NeedsDelay() ? 1 : 0;
```

**Review by yiwu-arbug — COMMENTED** [review]
Rest LGTM.

**Review by yiwu-arbug — APPROVED** [review]

**Review by tabokie — APPROVED** [review]

**Pull request merged**